tests/unit_tests/test_map_pool.py

```diff
@@ -31,7 +33,45 @@ def test_choose_map(map_pool_factory):
     # Make the probability very low that the test passes because we got lucky
     for _ in range(20):
         chosen_map = map_pool.choose_map([1, 2, 3])
-        assert chosen_map == (4, "CHOOSE_ME", "maps/choose_me.v001.zip")
+        assert chosen_map == (4, "CHOOSE_ME", "maps/choose_me.v001.zip", 1)
+
+
+def test_choose_map_with_weights(map_pool_factory):
+    map_pool = map_pool_factory(maps=[
+        Map(1, "some_map", "maps/some_map.v001.zip", 1),
+        Map(2, "some_map", "maps/some_map.v001.zip", 1),
+        Map(3, "some_map", "maps/some_map.v001.zip", 1),
+        Map(4, "CHOOSE_ME", "maps/choose_me.v001.zip", 1000000),
+    ])
+
+    # Make the probability very low that the test passes because we got lucky
+    for _ in range(20):
+        chosen_map = map_pool.choose_map()
+        assert chosen_map == (4, "CHOOSE_ME", "maps/choose_me.v001.zip", 1000000)
```

Comment:
Hm, interesting. Kindof a hard thing to test I guess. I'm just looking forward to that one time I get super unlucky and it fails on a release build :P

Reply:
Yeah I was considering if I should even have this or make it that m out of n should be chosen

Reply:
I think if the probability of it failing is sufficiently small then it will still be ok. Like 10^-6 or something idk.

I guess the probability that this fails is already pretty close to that:
```
(1 - (1000000/1000003)) * 20 = 6e-05
```